### PR TITLE
bpo-30455: Generate tokens related C code and docs from token.py.

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
https://bugs.python.org/issue30455

### CODE REVIEW

**Comment by matrixise** [comment]
@serhiy-storchaka if `Parser/token.c` is auto-generated by the tool, why do you include it in this PR ?

File: Doc/library/token.rst [hunk 1]
```diff
@@ -1,118 +1,64 @@
...
-      Added :data:`AWAIT` and :data:`ASYNC` tokens. Starting with
-      Python 3.7, "async" and "await" will be tokenized as :data:`NAME`
-      tokens, and :data:`AWAIT` and :data:`ASYNC` will be removed.
-
-   .. versionchanged:: 3.7
-      Added :data:`COMMENT`, :data:`NL` and :data:`ENCODING` to bring
-      the tokens in the C code in line with the tokens needed in
-      :mod:`tokenize` module. These tokens aren't used by the C tokenizer.
\ No newline at end of file
+.. Auto-generated by Tools/scripts/generate_token_rst.py
```
Comment (berkerpeksag) [inline Doc/library/token.rst]: Can't we use the include directive to include the token list?

File: Doc/library/token.rst [hunk 2]
```diff
@@ -1,118 +1,64 @@
...
-.. module:: token
-   :synopsis: Constants representing terminal nodes of the parse tree.
-
-.. sectionauthor:: Fred L. Drake, Jr. <fdrake@acm.org>
-
-**Source code:** :source:`Lib/token.py`
-
---------------
-
-This module provides constants which represent the numeric values of leaf nodes
```
Comment (berkerpeksag) [inline Doc/library/token.rst]: We are going to keep the existing documentation, right?

File: Doc/library/token.rst (hunk 1 above)
Comment (serhiy-storchaka) [reply Doc/library/token.rst]: Oh, yes, it was the intention. But due to an error in Makefile `token.rst` was rewritten by the content of generated `token-list.inc`. I need to restore the original `token.rst` with the include directive.

**Comment by markshannon** [comment]
Nice cleaning up of the code. Generally LGTM.

Tools/scripts/generate_token_h.py and Tools/scripts/generate_token_rst.py are very similar, it would be neater to merge them.

**Comment by serhiy-storchaka** [comment]
Created alternate PR #10370 which takes all information from Grammar/Tokens and use a single generating script.

**Comment by serhiy-storchaka** [comment]
Closed in favor of #10370.

**Pull request closed without merging**